File: Sheet 4 (Loops 2)/I.cpp
#include <iostream>
#include <algorithm>
using namespace std;
int main()
{
    int arr[5] , c = 0 , ans = -1;
    for (int i = 0; i < 4 ;i ++){
        cin >> arr[i];
    }
    for (int i = 0 ; i < 4 ;i ++){
        for(int j = i+1 ; j < 4 ; j++){
            if (arr[i] == arr[j]){
                c++;
                arr[j]= ans;
                ans--;
            }
        }
    }
    cout << c;
        return 0;
}

File: Sheet 7 (STLs I)/I.py
tc = int(input())

for i in range(tc):
    n = int(input())
    arr = list(map(int, input().split()))
    
    used = set()
    cur_largest = n
    temp = []
    ans = []
    
    for i in range(n-1, -1, -1):
        temp.append(arr[i])
        used.add(arr[i])
        if arr[i] == cur_largest:
            while temp:
                ans.append(temp.pop())
            while cur_largest in used:
                cur_largest -= 1
    print(*ans)
